lib/datamodel/graph.js

```diff
@@ -0,0 +1,79 @@
+/** Class that signifies a graph */
+class Graph {
+    /**
+      * The underlying graph object. This is an object with nodes and edges.
+      * The class contains an adjacency list property called graph that represents the edges
+      * between each of the nodes. Each edge can have attributes such as weight (which 
+      * by default is 10) and whether or not the edge is directed.
+      *
+      * */ 
+    constructor() {
+       this.graph = new Map()
+    }
+ 
+    clear() {
+       this.graph = new Map()
+       return true
+    }
+ 
+    addNode(node) {
+       if (!this.graph.has(node)) {
+          this.graph.set(node, new Map())
+       }
+    }
+ 
+    addNodes(nodes) {
+       if (!Array.isArray(nodes)) {
+          throw new Error('Parameter to addNodes is not an array')
+      }
```

Comment:
Rather than check if it's an array. Can you check if it's iterable? Then you can use the for ... of statement instead.